src/python/pants/testutil/pants_run_integration_test.py

```diff
@@ -557,42 +558,83 @@ def file_renamed(self, prefix, test_name, real_name):
             os.rename(real_path, test_path)
 
     @contextmanager
-    def temporary_file_content(self, path, content, binary_mode=True):
+    def temporary_directory_literal(
+        self, path: Union[str, Path],
+    ):
```

Comment:
```suggestion
    ) -> Iterator[None]:
```

This type hint will make sure call sites don't do something like `temporary_directory_literal() as foo`